splink/linker.py

```diff
@@ -2422,3 +2565,71 @@ def estimate_probability_two_random_records_match(
             " possible comparisons, we expect a total of around "
             f"{num_expected_matches:,.2f} matching pairs"
         )
+
+    def invalidate_cache(self):
+        """Invalidate the Splink cache.  Any previously-computed tables
+        will be recomputed.
+        This is useful, for example, if the input data tables have changed.
+        """
+        # Before Splink executes a SQL command, it checks the cache to see
+        # whether a table already exists with the name of the output table
+
+        # This function has the effect of changing the names of the output tables
+        # to include a different unique id
+
+        # As a result, any previously cached tables will not be found
+        self._cache_uid = ascii_uuid(8)
+
+        # As a result, any previously cached tables will not be found
+        self._intermediate_table_cache.invalidate_cache()
+
+        # Also drop any existing splink tables from the database
+        # Note, this is not actually necessary, it's just good housekeeping
+        self._delete_tables_created_by_splink_from_db()
+
+    def register_table_input_nodes_concat_with_tf(self, input_data, overwrite=False):
+        """Register a pre-computed version of the input_nodes_concat_with_tf table that
+        you want to re-use e.g. that you created in a previous run
+
```

Comment:
In this method name and description there is a different table-name than that used behind-the-scenes - is this deliberate? I know the other one is 'internal use', but initially spent a moment trying to work out if it was the same thing.

Reply:
Good spot, I should have mentioned this.  

I want this method name to be as clear as possible.  I actually want to go through all the internal names to make them clearer/more descriptive, but i haven't got around to doing that yet